maintenance/health/servicehealthcheck/healthchecker.go

```diff
@@ -68,13 +74,55 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// Does all HealthChecks and lists the result in the response body
+func checkAll(w http.ResponseWriter) {
+	results := make(map[string]error)
+	// do all checks and save the results
+	checks.Range(func(key, value interface{}) bool {
```

Comment:
As far as I can see, this is basically a generalized implementation of the singular healthCheck.
In both cases you use (name string, hc HealthCheck) in order to check for initialization errors and health status.
You could extract this loop and move it to a separate function, and treat singular healthChecks as a map with a size of 1.
That would eliminate the need for duplicate code.

Reply:
At least you should separate the health check logic from the response generation.